packages/lib-classifier/src/store/FieldGuideStore.js

```diff
@@ -0,0 +1,113 @@
+    showModal: types.optional(types.boolean, false),
+    type: types.optional(types.string, 'field_guides')
+  })
+
+  .actions(self => {
+    function afterAttach() {
+      createProjectObserver()
+    }
+
+    function createProjectObserver() {
+      const projectDisposer = autorun(() => {
+        const project = getRoot(self).projects.active
+        if (project) {
+          self.reset()
+          self.fetchFieldGuide()
+        }
+      })
+      addDisposer(self, projectDisposer)
+    }
+
+    function reset () {
+      self.active = undefined
+      self.resources.clear()
+      self.activeMedium = undefined
+      self.attachedMedia.clear()
+      self.activeItem = undefined
+      self.showModal = false
+    }
+
+    // TODO: this might need to paginate for field guides that have 20+ items
+    const fetchMedia = flow(function * fetchMedia (fieldGuide) {
+      const { type } = self
+      const client = getRoot(self).client.panoptes
+      if (fieldGuide) {
+        self.loadingState = asyncStates.loading
+        try {
+          const url = `/${type}/${fieldGuide.id}/attached_images`
+          const response = yield client.get(url)
+          const { media } = response.body
+          if (media && media.length > 0) self.setMediaResources(media)
+        } catch (error) {
+          // We're not setting the store state to error because
+          // we do not want to prevent the field guide from rendering
+          console.error(error)
+        }
+      }
+    })
+
+    function setMediaResources (media) {
+      media.forEach(medium => self.attachedMedia.put(medium))
+    }
+
+    // TODO: move req in panoptes.js
+    function * fetchFieldGuide () {
+      const { type } = self
+      const project = getRoot(self).projects.active
+      const client = getRoot(self).client.panoptes
+
+      self.loadingState = asyncStates.loading
+      try {
+        const response = yield client.get(`/${type}`, { project_id: project.id })
+        const fieldGuide = response.body[type][0]
+        if (fieldGuide) {
+          yield fetchMedia(fieldGuide)
+          self.setResource(fieldGuide)
+          self.setActive(fieldGuide.id)
+          self.loadingState = asyncStates.success
+        } else {
+          self.loadingState = asyncStates.success
+        }
+      } catch (error) {
+        console.error(error)
+        self.loadingState = asyncStates.error
+      }
+    }
+
+    function setModalVisibility (boolean) {
+      self.showModal = boolean
+    }
+
+    function setActiveItem (index) {
+      const fieldGuide = self.active
+      if (fieldGuide && fieldGuide.items.length === index + 1 && fieldGuide.items[index]) {
+        self.activeItem = index
+        if (fieldGuide.items[index].icon) self.activeMedium = fieldGuide.items[index].icon
+      }
+    }
+
+    return {
+      afterAttach,
+      fetchFieldGuide: flow(fetchFieldGuide),
+      reset,
+      setActiveItem,
+      setMediaResources,
+      setModalVisibility
+    }
```

Comment:
Are `afterAttach`, `setActiveItem` and `setMediaResources` ever accessed from outside the model? If not, you could remove them from the export, which would simplify testing the model, and reduce the amount of stubbing the tests need to do.

Reply:
`setActiveItem` definitely will be used by a click handler. The others probably not. Is it ok to update in the PR for the UI?

Reply:
Since it's approved. I'll just plan on updating the exported actions in the next PR. Thanks!